source/utils/network.ts

```diff
@@ -13,19 +15,30 @@ export const isActiveNetwork = (chain: string, chainId: number) => {
 /**
  * `{ chaindId, url }` is compatible with `INetwork`
  */
-export const isSysNetwork = ({
+export const networkChain = async (network: { chainId: number; url: string }) =>
+  (await isBitcoinBasedNetwork(network)) ? 'syscoin' : 'ethereum';
+
+/**
+ * `{ chaindId, url }` is compatible with `INetwork`
+ */
+export const isBitcoinBasedNetwork = async ({
   chainId,
   url,
 }: {
   chainId: number;
   url: string;
 }) => {
-  const { networks } = store.getState().vault;
-  return Boolean(networks.syscoin[chainId]) && url.includes('blockbook');
-};
+  try {
+    const { networks } = store.getState().vault;
 
-/**
- * `{ chaindId, url }` is compatible with `INetwork`
- */
-export const networkChain = (network: { chainId: number; url: string }) =>
-  isSysNetwork(network) ? 'syscoin' : 'ethereum';
+    const isSyscoinChain = Boolean(networks.syscoin[chainId]);
+
+    if (!isSyscoinChain) return false;
+
+    const { valid } = await validateSysRpc(url);
```

Comment:
opinion: Is this validation needed? This call will cause delay and might throw some unwanted errors on a fn that is 'simple'